### ISSUE TITLE
Client does not display error message when server returns 503

### ISSUE THREAD

**Issue by dpursehouse** [github/git-lfs#1486]
Using the current head of the master branch:

```
git-lfs $ git describe
v1.4.0-9-g89f37f4
```

When the server returns 503 with a message `LFS is not available`, the client does not display the message:

```
test-project $ git push origin HEAD:refs/for/master
Git LFS: (0 of 3 files) 0 B / 1.31 MB

Errors logged to /Users/dpursehouse/temp/test-project/.git/lfs/objects/logs/20160826T144116.556014811.log
Use `git lfs logs last` to view the log.
error: failed to push some refs to 'ssh://admin@localhost:29418/test-project'
```

The user has to use `git lfs logs last` to see the message:

```
[snip]
batch response: Fatal error: http: LFS is not available
[snip]
```

This seems to be a recent change in behaviour.  With version 1.3.1 the error was displayed:

```
test-project $ git push origin HEAD:refs/for/master
Git LFS: (0 of 3 files) 0 B / 1.31 MB
LFS is not available

Errors logged to /Users/dpursehouse/temp/test-project/.git/lfs/objects/logs/20160826T145330.6359595.log
Use `git lfs logs last` to view the log.
error: failed to push some refs to 'ssh://admin@localhost:29418/test-project'
```

**Comment by technoweenie** [comment]
Thanks for the report, looks like a regression with the new errors package.

**Comment by ttaylorr** [comment]
👋  @dpursehouse, thanks for opening this.

Just to clarify, are you talking about the v1.3 batch API wherein an error is reported through the [batch response](https://github.com/github/git-lfs/blob/master/docs/api/v1.3/http-v1.3-batch-response-schema.json#L54-L63)?

I set up a branch to try and replicate this, but was unable to reproduce. In [`git-lfs/503-no-report`](https://github.com/github/git-lfs/compare/503-no-report?expand=1), the error was reported as follows:

```
$ git lfs push origin master
Git LFS: (1 of 1 files, 1 skipped) 2 B / 2 B, 16 B skipped                     
   [432cf07eb9a1b9d66f3009f001064c2466c237015f64ac05177b835d7effd138] welp: [503] welp
   error: failed to push some refs to 'http://127.0.0.1:50160/test-push-failures-status-batch-503'
```

The error traces itself through the code path as follow:

- The `test/test-push-failures.sh` file [generates a file `bad.dat` with a flag](https://github.com/github/git-lfs/blob/4963c30f93548e444955eb5da1643e3a8857bdbd/test/test-push-failures.sh#L16) to signal to the gitserver that it should [detect the appropriate "handler"](https://github.com/github/git-lfs/blob/4963c30f93548e444955eb5da1643e3a8857bdbd/test/cmd/lfstest-gitserver.go#L369) and send the [appropriate response](https://github.com/github/git-lfs/blob/4963c30f93548e444955eb5da1643e3a8857bdbd/test/cmd/lfstest-gitserver.go#L386-L387).
- That error gets [picked up by the transfer queue](https://github.com/github/git-lfs/blob/4963c30f93548e444955eb5da1643e3a8857bdbd/lfs/transfer_queue.go#L370) and reported to `q.errorc`.
- [The transfer queue works in the background to collect errors](https://github.com/github/git-lfs/blob/4963c30f93548e444955eb5da1643e3a8857bdbd/lfs/transfer_queue.go#L410-L416).
- and that worker is [waited upon](https://github.com/github/git-lfs/blob/ddfaae3c3a5ada9a0afe4c24c596b662bbc28ffc/commands/uploader.go#L143) by the `commands.upload` function.
- All errors are then [collected](https://github.com/github/git-lfs/blob/ddfaae3c3a5ada9a0afe4c24c596b662bbc28ffc/commands/uploader.go#L145-L147) and reported.

**Comment by ttaylorr** [comment]
Could you give this a shot with `GIT_CURL_VERBOSE=1` and see what you get?

**Comment by dpursehouse** [comment]
@ttaylorr  Here's the verbose output:

```
test-project $ GIT_CURL_VERBOSE=1 GIT_TRACE=1 git push origin HEAD:refs/for/master
14:43:33.485842 git.c:350               trace: built-in: git 'push' 'origin' 'HEAD:refs/for/master'
14:43:33.487241 run-command.c:336       trace: run_command: 'ssh' '-p' '29418' 'admin@localhost' 'git-receive-pack '\''/test-project'\'''
14:43:33.520178 run-command.c:336       trace: run_command: '.git/hooks/pre-push' 'origin' 'ssh://admin@localhost:29418/test-project'
14:43:33.525169 git.c:563               trace: exec: 'git-lfs' 'pre-push' 'origin' 'ssh://admin@localhost:29418/test-project'
14:43:33.525724 run-command.c:336       trace: run_command: 'git-lfs' 'pre-push' 'origin' 'ssh://admin@localhost:29418/test-project'
trace git-lfs: run_command: 'git' version
trace git-lfs: run_command: git rev-list --objects --stdin --
trace git-lfs: run_command: git cat-file --batch-check
trace git-lfs: run_command: git cat-file --batch
trace git-lfs: run_command: 'git' config -l
trace git-lfs: tq: running as batched queue, batch size of 100
trace git-lfs: tq: sending batch of size 3
trace git-lfs: api: batch 3 files
trace git-lfs: HTTP: POST http://admin@localhost:8080/test-project/info/lfs/objects/batch
> POST /test-project/info/lfs/objects/batch HTTP/1.1
> Host: localhost:8080
> Accept: application/vnd.git-lfs+json; charset=utf-8
> Content-Type: application/vnd.git-lfs+json; charset=utf-8
> User-Agent: git-lfs/1.4.0 (GitHub; darwin amd64; go 1.6.3; git 32be580)
>
{"operation":"upload","objects":[{"oid":"ba0ef3cdb015b98897f1efa7de51cb97d8370c2fee320a2ff7f377ea551ad958","size":1322764},{"oid":"4fa3fb84d8cb8acf1624fef16e1441e9a49ad7ffb3f165e46fda009f2c14d396","size":23658},{"oid":"2ee6441ad823c2be85a3a86480d8b096dd3c1b3c2a4fb2969203bb57e9caf6b8","size":30700}]}trace git-lfs: HTTP: 503


< HTTP/1.1 503 Service Unavailable
< Transfer-Encoding: chunked
< Content-Type: application/vnd.git-lfs+json; charset=utf-8
< Date: Fri, 26 Aug 2016 05:43:33 GMT
<
trace git-lfs: HTTP: {"message":"LFS is not available"}
{"message":"LFS is not available"}trace git-lfs: HTTP:
trace git-lfs: api error: Fatal error: http: LFS is not available
Git LFS: (0 of 3 files) 0 B / 1.31 MB

Errors logged to /Users/dpursehouse/temp/test-project/.git/lfs/objects/logs/20160826T144335.982103126.log
Use `git lfs logs last` to view the log.
error: failed to push some refs to 'ssh://admin@localhost:29418/test-project'
```

This is with the v1 API:

```
version https://git-lfs.github.com/spec/v1
```

It's not with the reference server implementation, but through Gerrit Code Review with the API in JGit.

The change in behaviour only shows when using the latest client version.  It was OK in 1.3.2.

**Comment by ttaylorr** [comment]
:wave: @dpursehouse. Sorry for not getting back to you sooner. I had this email sitting in my inbox all of last week, but was participating in our internal Hack Week and didn't respond as quickly as I would have liked.

I took a deeper look into this, and all is well in terms of LFS internals. The issue, I found, lies in how we report errors back to the client in [`commands/commands.go`](https://github.com/github/git-lfs/blob/a06accadc8a4bcd6a9f225f8d5eb3806c393b7d2/commands/commands.go#L80-L84). I went ahead and fixed the bug in #1505.

Thanks again for reporting this!
